multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -77,20 +84,22 @@ func NewMemberClusterSetReconciler(client client.Client,
 func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	clusterSet := &multiclusterv1alpha1.ClusterSet{}
 	err := r.Get(ctx, req.NamespacedName, clusterSet)
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
+	r.commonAreaLock.Lock()
+	defer r.commonAreaLock.Unlock()
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			return ctrl.Result{}, err
 		}
 		klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
-		stopErr := r.remoteCommonAreaManager.Stop()
-		r.remoteCommonAreaManager = nil
-		r.clusterSetConfig = nil
-		r.clusterID = common.InvalidClusterID
-		r.clusterSetID = common.InvalidClusterSetID
-
-		return ctrl.Result{}, stopErr
+		if r.remoteCommonArea != nil {
+			stopErr := r.remoteCommonArea.Stop()
+			r.remoteCommonArea = nil
```

Comment:
we should not set it to nil at error?

Reply:
After double check the `stop()`, it's calling context's cancel function, so there is actually no error returned. I removed the return error on `stop()`.